include/deal.II/matrix_free/shape_info.h

```diff
@@ -101,6 +101,29 @@ namespace internal
 
 
 
+    template <typename T>
+    struct UnivariateShapeDataTrait;
+
+    template <>
+    struct UnivariateShapeDataTrait<double>
+    {
+      using value_type = double;
+    };
+
+    template <>
+    struct UnivariateShapeDataTrait<float>
+    {
+      using value_type = float;
+    };
+
+    template <typename T, std::size_t width>
+    struct UnivariateShapeDataTrait<VectorizedArray<T, width>>
+    {
+      using value_type = T;
+    };
```

Comment:
@kronbichler Do we have somewhere a functionality like this?

Reply:
No, I don't think so, but it would be useful to have in a central place (read: `vectorization.h`). And in general, I think you can simplify this by saying
```c++
template <typename T>
struct UnivariateShapeDataTrait
{
  using value_type = T;
};
```
and then only a specialization for `VectorizedArray<T, width>`.